fasort.sh

```diff
@@ -2,21 +2,21 @@
 # Author: Noxy (a dhole who likes paws)
 # This script will find and move any FA files in the working directory and move them to artist folders under $SORTDIR.
 
+set -euo pipefail
+
 # Directory where you want files to end up
 SORTDIR=~/Pictures/FA_Sorted
 
 # Find files in pwd that match what I think is a decent regex for FA image files
-for f in `ls | egrep '^\d+\.[0-9a-zA-Z-]+_.+\.\w+$'`
+find . -iregex '.*/[0-9]+\.[a-z0-9]+_.*' -print0 | while read -d $'\0' f
 do
 
     # awk out the artist name
-    ARTIST=$(echo $f | awk -F '.' '{print$2}' | awk -F _ '{print$1}')
+    ARTIST="$(printf '%s' "$(basename "$f")" | awk -F '.' '{print$2}' | awk -F _ '{print$1}')"
```

Comment:
All variables when used as an argument for a command have to be quoted, otherwise spaces in the string will be treated as separate arguments to the command, for example this code:

```
var="a b"
mkdir $var
```
would create two directories called `a` and `b` and this:

```
var="a b"
mkdir "$var"
```
would create a directory called `a b` (with a space)

also using `echo` to pipe a variable to a command is problematic because `echo` can take some switches which could be present in the variable you're trying to `echo`, for example, this

```
var="-n"
echo "$var"
```
would not echo anything, because `-n` is a switch for the `echo` command. `printf '%s' "$var"` is the safest option